js/__internal/viz/chart_components/m_advanced_chart.ts

```diff
@@ -0,0 +1,863 @@
+          pane,
+          priority,
+          optionPath,
+          crosshairMargin: rotated ? crosshairMargins.y : crosshairMargins.x,
+        }, rotated) as never);
+      });
+    });
+
+    that._redesignAxes(argumentAxesPopulatedOptions, true, paneWithNonVirtualAxis);
+    that._redesignAxes(valueAxesPopulatedOptions, false);
+  },
+
+  _redesignAxes(options, isArgumentAxes, paneWithNonVirtualAxis) {
+    const that = this;
+    const axesBasis = [];
+    let axes = isArgumentAxes ? that._argumentAxes : that._valueAxes;
+
+    _each(options, (_, opt) => {
+      const curAxes = axes && axes.filter((a) => a.name === opt.name
+                && (!_isDefined(opt.pane) && that.panes.some((p) => p.name === a.pane) || a.pane === opt.pane));
+      if (curAxes && curAxes.length > 0) {
+        _each(curAxes, (_, axis) => {
+          const axisTypes = getAxisTypes(that._groupsData, axis, isArgumentAxes);// T891599
+          axis.updateOptions(opt);
+          if (isArgumentAxes) {
+            axis.setTypes(axisTypes.argumentAxisType, axisTypes.argumentType, 'argumentType');
+          } else {
+            axis.setTypes(axisTypes.valueAxisType, axisTypes.valueType, 'valueType');
+          }
+          axis.validate();
+          axesBasis.push({ axis } as never);
+        });
+      } else {
+        axesBasis.push({ options: opt } as never);
+      }
+    });
+
+    if (axes) {
+      _reverseEach(axes, (index, axis) => {
+        if (!axesBasis.some((basis: any) => basis.axis && basis.axis === axis)) {
+          that._disposeAxis(index, isArgumentAxes);
+        }
+      });
+    } else if (isArgumentAxes) {
+      axes = that._argumentAxes = [];
+    } else {
+      axes = that._valueAxes = [];
+    }
+
+    _each(axesBasis, (_, basis: any) => {
+      let { axis } = basis;
+      if (basis.axis && isArgumentAxes) {
+        basis.axis.isVirtual = basis.axis.pane !== paneWithNonVirtualAxis;
+      } else if (basis.options) {
+        axis = that._createAxis(
+          isArgumentAxes,
+          basis.options,
+          isArgumentAxes ? basis.options.pane !== paneWithNonVirtualAxis : undefined,
+        );
+        axes.push(axis);
+      }
+      axis.applyVisualRangeSetter(that._getVisualRangeSetter());
+    });
+  },
+
+  _disposeAxis(index, isArgumentAxis) {
+    const axes = isArgumentAxis ? this._argumentAxes : this._valueAxes;
+    const axis = axes[index];
+
+    if (!axis) return;
+
+    axis.dispose();
+    axes.splice(index, 1);
+  },
+
+  _disposeAxes() {
+    const that = this;
+    const disposeObjectsInArray = that._disposeObjectsInArray;
+    disposeObjectsInArray.call(that, '_argumentAxes');
+    disposeObjectsInArray.call(that, '_valueAxes');
+  },
+
+  _appendAdditionalSeriesGroups() {
+    this._crosshairCursorGroup.linkAppend();
+    // this._legendGroup.linkAppend();
+    this._scrollBar && this._scrollBarGroup.linkAppend(); // TODO: Must be appended in the same place where removed (chart)
```

Comment:
What does this TODO mean? Shouldnt we move it to a card?
Is the comment above necessary?
```suggestion
    this._scrollBarGroup?.linkAppend();
```